### PR TITLE
AP_MSP: new wiki entry for MSP support

### PR DESCRIPTION
**Pull request by yaapu**
Basic new wiki entry for the MSP protocol and DJI MSP

Still missing:
 - integration in the wiki
 - MSP rangefinder setup
 - MSP Optical flow setup

### CODE REVIEW

**Comment by Hwurzburg** [comment]
@yaapu Alex, starting to work on this.....questions:
1. does the python osd emulator support all present OSD panels? 
2. are future osd panels automatically availabe like on the internal OSD? on goggles? on emulator?
3. is the limitation of only one panel, just an emulator issue? what about fonts and units? are they all supported on DJI goggles?

**Comment by Hwurzburg** [comment]
Pushed some fixe, changes,  and expansions.....need list of supported panel items and I assume they are also supported in the emulator, correct?

**Comment by yaapu** [comment]
> @yaapu Alex, starting to work on this.....questions:
> 
> 1. does the python osd emulator support all present OSD panels?
> 2. are future osd panels automatically availabe like on the internal OSD? on goggles? on emulator?
> 3. is the limitation of only one screen, just an emulator issue? what about fonts and units? are they all supported on DJI goggles?

1. blheli32 esc temperature is missing as well as sidebars and crosshair
2. No, OSD panels are rendered by the goggles we do not draw on the goggles screen, we provide telemetry and positions and the goggles render them. Adding new panels requires new firmware from DJI that supports them
3. no, was my choice for the first version, @tridge suggested a feature freeze to let the PR make into master. Adding more screens is possible, screen switching would be identical as integrated OSD, DJI goggles do not support multiple screens it would be an ArduPilot thing to drive it

**Comment by Hwurzburg** [comment]
thanks added the list of supported panels...please review....

**Comment by yaapu** [comment]
- Metric and Imperial units are both supported, you get m/s or ft/s, don't remember what you get for distances.
If you select metric in ardupilot you get metric, if you select any other unit you get imperial on the DJI goggles.

**Comment by Hwurzburg** [comment]
fixed....added note re emulation not supporting unit change or multiple screens

**Comment by yaapu** [comment]
@Hwurzburg about consuming MSP sensor messages, AP_periph now can send MSP messages for GPS/BARO/MAG and soon Airspeed and ArduPilot can consume them, so a single uart connected to a matek can gps can provide all 4 sensors

File: common/source/docs/common-msp-osd-overview.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+ .. image:: ../../../images/msp_dji_goggles_re.jpeg
+    :target: ../_images/msp_dji_goggles_re.jpeg
+ 
+ArduPilot supports the special MSP OSD protocol which allows displaying flight data on the DJI goggles, much like with external MAVLink OSDs or the internal integrated OSD in many flight controllers.
+
+Features
+========
+ 
+ - ArduPilot currently supports many, but not all, of the OSD panel items as noted in the section below.
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview.rst:18]: Not sure what to change here...we list all the panels we support from ArduPilot, not sure we should mention we support all panels as that could go out of date without us knowing...if new DJI panesl get added and WE add them to our support we can update the list below

File: common/source/docs/common-msp-osd-overview.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+ - Changing display units other than metric are not currently supported.
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview.rst]: fixed in last push

File: common/source/docs/common-msp-overview.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+==============================
+
+ArduPilot supports the MSP protocol for telemetry and sensors via any of its serial ports. This allows ArduPilot to send its telemetry data to MSP compatible devices, such as DJI goggles, for On Screen Display (OSD).
+
+Protocol overview
+-----------------
+
+MSP is the main communication protocol used by all Betaflight derived flight stacks.
+It's a binary message based protocol used for control, telemetry and sensors.
+ArduPilot's MSP protocol module is ported from Betaflight and iNav, and supports both MSPV1 and MSPV2. It also supports iNav's MSPV2 sensor messages for lidar and optical flow.
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-msp-overview.rst:14]: added as futures

File: common/source/docs/common-msp-osd-overview.rst [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+- HDOP
+- Flightime
+- Clock
+- Gps_latitude
+- Gps_longitude
+- Distance Traveled
+- Statistics
+- Climb Efficiency
+- Efficiency
+
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview.rst:72]: updated

File: common/source/docs/common-msp-osd-overview.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (yaapu) [reply common/source/docs/common-msp-osd-overview.rst:18]: ...yes, we never mention firmware versions in other wiki pages so probably your approach is the right one, we provide a list and eventually update it

File: common/source/docs/common-msp-osd-overview.rst [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+Configuration
+=============
+
+To enable MSP OSD, set the following parameters (using SERIAL port 2 as the port to attach to the DJI Air unit using both TX and RX lines):
+
+ - :ref:`OSD_TYPE<OSD_TYPE>` = 3
+ - :ref:`SERIAL2_PROTOCOL<SERIAL2_PROTOCOL>` = 33
+ - :ref:`MSP_OPTIONS<MSP_OPTIONS>` = 0 (polling mode)
+
+
```
Comment (yaapu) [inline common/source/docs/common-msp-osd-overview.rst:32]: We might add a note that external MSP osd has to be enabled on the goggles as well:
Under the DJI goggle’s SETTINGS, DISPLAY, CUSTOM OSD, you can turn the OSD ON or OFF.

**Comment by Hwurzburg** [comment]
going to remove the landing page additions...I've added them in the Upcoming features section...will merge this....the new pages will be in the wiki...but links will only appear in the upcoming features section...

File: common/source/docs/common-msp-osd-overview.rst (hunk 5 above)
Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview.rst:32]: will do

Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview.rst:32]: will do

**Comment by Hwurzburg** [comment]
merging...thanks!!

**Comment by vierfuffzig** [comment]
@Hwurzburg what do you think about using sim_vehicle.py --osdmsp command? imho it makes running the HD osd simulation a lot easier.

**Comment by Hwurzburg** [comment]
yeah...forgot to update all that...AND make a new video....will do all that ....thanks for the reminder...

**Comment by Hwurzburg** [comment]
done, will update the video tomorrow

**Pull request merged**